lib/host/wasi/inode-win.cpp

```diff
@@ -57,6 +68,347 @@ createNullTerminatedString(std::string_view View) noexcept {
+    break;
+  case ERROR_IO_INCOMPLETE:
+  case ERROR_OPERATION_ABORTED:
+    Error = __WASI_ERRNO_INTR;
+    break;
+  case ERROR_META_EXPANSION_TOO_LONG:
+    Error = __WASI_ERRNO_2BIG;
+    break;
+  case ERROR_NEGATIVE_SEEK:
+  case ERROR_SEEK_ON_DEVICE:
+    Error = __WASI_ERRNO_SPIPE;
+    break;
+  case ERROR_NOT_SAME_DEVICE:
+    Error = __WASI_ERRNO_XDEV;
+    break;
+  case ERROR_SHARING_BUFFER_EXCEEDED:
+    Error = __WASI_ERRNO_NFILE;
+    break;
+  case ERROR_TOO_MANY_MODULES:
+  case ERROR_TOO_MANY_OPEN_FILES:
+    Error = __WASI_ERRNO_MFILE;
+    break;
+  case ERROR_WAIT_NO_CHILDREN:
+    Error = __WASI_ERRNO_CHILD;
+    break;
+  case ERROR_WRITE_PROTECT:
+    Error = __WASI_ERRNO_ROFS;
+    break;
+  case ERROR_CANT_RESOLVE_FILENAME:
+    Error = __WASI_ERRNO_LOOP;
+    break;
+  case ERROR_CONNECTION_ABORTED:
+    Error = __WASI_ERRNO_CONNABORTED;
+    break;
+  case ERROR_CONNECTION_REFUSED:
+    Error = __WASI_ERRNO_CONNREFUSED;
+    break;
+  case ERROR_HOST_UNREACHABLE:
+    Error = __WASI_ERRNO_HOSTUNREACH;
+    break;
+  case ERROR_INVALID_HANDLE:
+    Error = __WASI_ERRNO_BADF;
+    break;
+  case ERROR_NETNAME_DELETED:
+    Error = __WASI_ERRNO_CONNRESET;
+    break;
+  case ERROR_NETWORK_UNREACHABLE:
+    Error = __WASI_ERRNO_NETUNREACH;
+    break;
+  case ERROR_NOT_CONNECTED:
+    Error = __WASI_ERRNO_NOTCONN;
+    break;
+  case ERROR_NOT_SUPPORTED:
+    Error = __WASI_ERRNO_NOTSUP;
+    break;
+  case ERROR_SEM_TIMEOUT:
+    Error = __WASI_ERRNO_TIMEDOUT;
+    break;
+  case ERROR_TOO_MANY_LINKS:
+    Error = __WASI_ERRNO_MLINK;
+    break;
+  default:
+    assumingUnreachable();
+  }
+  return Error;
+}
+
+constexpr DWORD attributeFlags(__wasi_oflags_t OpenFlags,
+                               __wasi_fdflags_t FdFlags) noexcept {
+  DWORD Flags = FILE_ATTRIBUTE_NORMAL;
+  if ((FdFlags & __WASI_FDFLAGS_NONBLOCK) != 0) {
+    Flags |= FILE_FLAG_OVERLAPPED;
+  }
+
+  // Source: https://devblogs.microsoft.com/oldnewthing/20210729-00/?p=105494
+  if ((FdFlags & __WASI_FDFLAGS_SYNC) || (FdFlags & __WASI_FDFLAGS_RSYNC)) {
+    // Linux does not implement O_RSYNC and glibc defines O_RSYNC as O_SYNC
+    Flags |= FILE_FLAG_WRITE_THROUGH | FILE_FLAG_NO_BUFFERING;
+  }
+  if (FdFlags & __WASI_FDFLAGS_DSYNC) {
+    Flags |= FILE_FLAG_WRITE_THROUGH;
+  }
+  if (OpenFlags & __WASI_OFLAGS_DIRECTORY) {
+    Flags |= FILE_ATTRIBUTE_DIRECTORY;
+  }
+
+  return Flags;
+}
+
+constexpr DWORD accessFlags(__wasi_fdflags_t FdFlags,
+                            uint8_t VFSFlags) noexcept {
+  DWORD Flags = 0;
+
+  if (VFSFlags & VFS::Read) {
+    if (VFSFlags & VFS::Write) {
+      Flags |= GENERIC_READ | GENERIC_WRITE;
+    } else {
+      Flags |= GENERIC_READ;
+    }
+  } else if (VFSFlags & VFS::Write) {
+    Flags |= GENERIC_WRITE;
+  }
+
+  if ((FdFlags & __WASI_FDFLAGS_APPEND) != 0) {
+    Flags |= FILE_APPEND_DATA;
+  }
+
+  return Flags;
+}
+/*
+  constexpr DWORD shareFlags(__wasi_oflags_t OpenFlags) {
```

Comment:
Do we need to keep this?

Reply:
It is fine if multiple processes can access ,read and write potentially at the same time? Currently all functions allow external processes to read/write a file opened by WasmEdge. If yes then we should remove it else we may have to modify it is some way.

Reply:
Yes, IIRC, we don't limit this. So it should be fine if there is any external process to access the file.

Reply:
Okay